buildpack/detect_test.go

```diff
@@ -118,10 +144,12 @@ func testDetect(t *testing.T, when spec.G, it spec.S) {
 		})
 
 		it("should fail if buildpacks have both a top level version and a metadata version", func() {
+			mockEnv.EXPECT().WithPlatform(platformDir).Return(append(os.Environ(), "ENV_TYPE=full"), nil)
```

Comment:
This always makes me sad... having to set `EXPECT` on the mock to make the test run even though it doesn't have anything to do with the expectation. That said I'm not sure if there's a way around it. Maaaybe we could make a variable called `someEnv` so that it reads `mockEnv.EXPECT().WithPlatform(platformDir).Return(append(os.Environ(), someEnv), nil)` so that it's a little clearer that we don't care about the env. Just a suggestion.

Reply:
I like it!

Resolved in latest commit.